slice-mapper/src/main/java/com/cognifide/slice/mapper/impl/processor/SliceResourceFieldProcessor.java

```diff
@@ -62,9 +71,57 @@ public Object mapResourceToField(Resource resource, ValueMap valueMap, Field fie
 								field.getName() });
 			}
 			return null;
-		} else {
-			return modelProvider.get(fieldType, nestedResource);
 		}
+		return modelProvider.get(fieldType, nestedResource);
+	}
+
+	private Object mapFollowUpResourceToField(Resource resource, ValueMap valueMap, Field field,
+			String propertyName) {
+
+		final Class<?> fieldType = field.getType();
+		Object value = valueMap.get(propertyName);
+
+		if (value == null) {
+			if (LOG.isDebugEnabled()) {
+				LOG.debug(
+						"the property [{}/{}] is undefined, assigning null value for [{}#{}]",
+						new Object[] { resource.getPath(), propertyName,
+								fieldType.getCanonicalName(), field.getName() });
+			}
+			return null;
+		}
+
+		if (!(value instanceof String)) {
+			if (LOG.isDebugEnabled()) {
+				LOG.debug(
```

Comment:
I'd increase log level here (warn), as this is an error in the content which might be difficult to trace and application code is expecting String value here.